### PR TITLE
Fix Docker container creation and error handling

### LINKED ISSUE #102: Docker unreachable error not reported by server on websocket connect
**Describe the bug**
When starting the server with Docker stopped, the server throws a backtrace error indicating Docker is not reachable instead of a clear error message.

**Steps to Reproduce**

1. Stop Docker.
2. Execute the command: `$ uvicorn opendevin.server.listen:app --reload --port 3000`.
3. Attempt to connect via WebSocket: `websocat ws://127.0.0.1:3000/ws`.
 

**Expected behavior**
The server should report a clear error message indicating Docker is down and a WebSocket connection cannot be established.

**Actual behavior**
The server starts and accepts the WebSocket connection, but upon attempting any operation that requires Docker, it crashes with a backtrace error pointing to Docker connectivity issues. The error log is as follows:
```
$ uvicorn opendevin.server.listen:app --reload --port 3000
INFO:     Will watch for changes in these directories: ['./OpenDevin-rbren/OpenDevin']
INFO:     Uvicorn running on http://127.0.0.1:3000 (Press CTRL+C to quit)
INFO:     Started reloader process [16775] using WatchFiles
INFO:     Started server process [16779]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     ('127.0.0.1', 49231) - "WebSocket /ws" [accepted]
INFO:     connection open
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/urllib3/connectionpool.py", line 793, in urlopen
    response = self._make_request(
               ^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/urllib3/connectionpool.py", line 496, in _make_request
    conn.request(
  File "/opt/homebrew/lib/python3.11/site-packages/urllib3/connection.py", line 400, in request
    self.endheaders()
  File "/opt/homebrew/Cellar/python@3.11/3.11.7_2/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py", line 1289, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/opt/homebrew/Cellar/python@3.11/3.11.7_2/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py", line 1048, in _send_output
    self.send(msg)
  File "/opt/homebrew/Cellar/python@3.11/3.11.7_2/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py", line 986, in send
    self.connect()
  File "/opt/homebrew/lib/python3.11/site-packages/docker/transport/unixconn.py", line 27, in connect
    sock.connect(self.unix_socket)
```

### PR DESCRIPTION
**Pull request by kvchitrapu**
1. Fixes #102. Print an error if Docker is not reachable. As docker restart tried in every session, server will automatically recover even if docker bounces.
2. Fixed some python module import order in `session.py` and `sandbox.py`

## Tests
When docker is not running:
```
INFO:     Started server process [58282]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     ('127.0.0.1', 59155) - "WebSocket /ws" [accepted]
INFO:     connection open
INFO:     Started server process [63469]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     ('127.0.0.1', 60186) - "WebSocket /ws" [accepted]
INFO:     connection open
Failed to stop container: Error while fetching server API version: ('Connection aborted.', PermissionError(13, 'Permission denied'))
Error creating controller. Please check Docker is running using `docker ps`.
Error! Error while fetching server API version: ('Connection aborted.', PermissionError(13, 'Permission denied'))```
```

When docker is running:
```
INFO:     ('127.0.0.1', 60325) - "WebSocket /ws" [accepted]
INFO:     connection open
INFO:     connection closed

==============
STEP 0
huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
To disable this warning, you can either:
        - Avoid using `tokenizers` before the fork if possible
        - Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)
```

### CODE REVIEW

File: opendevin/sandbox/sandbox.py [hunk 1]
```diff
@@ -178,7 +179,12 @@ def close(self):
         self.closed = True
 
     def stop_docker_container(self):
-        docker_client = docker.from_env()
+
+        try:
+            docker_client = docker.from_env()
+        except docker.errors.DockerException as e:
+            raise e
```
Comment (rbren) [inline opendevin/sandbox/sandbox.py:186]: should we print something here too?

File: opendevin/server/session.py [hunk 2]
```diff
@@ -130,7 +127,13 @@ async def create_controller(self, start_event=None):
         llm = LLM(model)
         AgentCls = Agent.get_cls(agent_cls)
         self.agent = AgentCls(llm)
-        self.controller = AgentController(self.agent, workdir=directory, callbacks=[self.on_agent_event])
+        try:
+            self.controller = AgentController(self.agent, workdir=directory, callbacks=[self.on_agent_event])
+        except Exception as e:
+            print("Error creating controller. Please check Docker is running using `docker ps`.")
+            print(f"Error! {e}", flush=True)
+            await self.send_error("Error creating controller. ")
```
Comment (rbren) [inline opendevin/server/session.py:135]: Might as well put the "Please check Docker" message here too

**Review by rbren — COMMENTED** [review]
Thanks for doing this!

File: opendevin/sandbox/sandbox.py (hunk 1 above)
Comment (kvchitrapu) [reply opendevin/sandbox/sandbox.py:186]: Implemented the change. The trace is Session > AgentController > DockerInteractive > restart_docker_container > stop_docker_container. Initially considered session.py, but it's more effective to advise on docker validation directly in stop_docker_container().

**Review by rbren — APPROVED** [review]

**Pull request merged**